### DISCUSSION TITLE
Cypress testing on mantine selects

### DISCUSSION THREAD

**Discussion by maiagrif** [mantinedev/mantine#4407] [Q&A]
Hello,

Has anyone had any luck with cypress testing on the Mantine selects?

We are incorporating this framework on our front-end, and I have yet to be able to chose a value from the dropdown menu for the Mantine Select in cypress testing. 

.select() does not work, and neither does .contain('value').click()

Any ideas?

**Comment by alwyn-ixo** [comment]
@maiagrif Hey there - I am now starting to use Cypress.io for testing our Next.js app.
Do you have any tips for the get() command?
Specifically, whether there is an existing equivalent to the `data-test` attribute?
PS: I will soon be looking at dropdown menus and will hopefully gain some insights about your question.

**Comment by rowellx68** [comment]
Here's a gist of a very basic custom command you could use with the `Select` component. There's a bit more work since you have to find the elements that Mantine renders.

https://gist.github.com/rowellx68/c94d305e5f4d32ae0a8eb80d597c23a0

Example usage would be

```ts
cy.get('[data-testid="select-favourite-number"]')
    .selectMantine('3')
    .should('have.value', '3')
 ```
 
 ```tsx
 <Select
   wrapperProps={{ 'data-testid': 'select-favourite-number' }}
   label="Favourite number"
   data={['1', '2', '3', '4']}
 />
 ```
